cert/cert.go

```diff
@@ -398,34 +434,34 @@ func (spec *Spec) Lifespan() time.Duration {
 	}
 
 	// This bit of code is necessary to confirm that the cert/key are older than the spec definition.
+	if spec.IsChangedOnDisk(spec.Key.Path) || spec.IsChangedOnDisk(spec.Cert.Path) {
+		// This is necessary to essentially force cfssl to regenerate since it's not spec aware.
+		log.Infof("refreshing due to spec %s having a newer mtime than key or cert", spec.Path)
+		spec.ResetLifespan()
+		return 0
+	}
+	return spec.tr.Lifespan()
+}
+
+func (spec *Spec) IsChangedOnDisk(path string) bool {
 	specStat, err := os.Stat(spec.Path)
 	if err != nil {
 		// The assertion here is that the spec actually
 		// exists. If it doesn't, something is wrong with the
 		// world.
-		panic("cert: certificate spec doesn't exist during RefreshKeys()")
+		panic("cert: isChangedOnDisk error")
```

Comment:
The previous message was slightly more descriptive, but likely often wrong. What about

```suggestion
		panic("cert: isChangedOnDisk: failed to stat certificate spec")
```